### PR TITLE
Jasmine data type

### PR DESCRIPTION
**Pull request by jasmineqyj**
Add `data_type.R` function and associated unit tests.

Issues:
#32 : function
#33 : unit test

### CODE REVIEW

File: tests/testthat.R [hunk 1]
```diff
@@ -0,0 +1,4 @@
+library(testthat)
+library(convertempr)
```
Comment (jasmineqyj) [inline tests/testthat.R]:
```suggestion
library(PrepR)
```

File: tests/testthat.R [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+test_check("prepR")
```
Comment (jasmineqyj) [inline tests/testthat.R]:
```suggestion
test_check("PrepR")
```

File: tests/testthat/test-data_type.R [hunk 3]
```diff
@@ -0,0 +1,15 @@
+library(testthat)
+library(tidyverse)
+library(prepR)
```
Comment (jasmineqyj) [inline tests/testthat/test-data_type.R]:
```suggestion
library(PrepR)
```

File: tests/testthat/test-data_type.R [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  test_df <- tibble(cat1 = c("apple", "pear"), cat2 = c(TRUE, FALSE), num1 = c(1L, 2L), num2 = c(2, 3.4))
+
+  expect_identical(data_type(test_df)$num, test_df %>% select(num1, num2))
+  expect_identical(data_type(test_df)$cat, test_df %>% select(cat1, cat2))
+})
+
+test_that("data_type correctly captures input error", {
+  expect_error(data_type("test_df")$num, "Please provide a valid data.frame object")
+  expect_error(data_type(tibble())$num, "Please provide a non-empty data.frame object")
+})
```
Comment (camadi) [inline tests/testthat/test-data_type.R:15]: Should your test be in a function?

```suggestion
test-data_type <- function(){
library(testthat)
library(tidyverse)
library(PrepR)

test_that("data_type correctly separates categorical and numeric", {
  test_df <- tibble(cat1 = c("apple", "pear"), cat2 = c(TRUE, FALSE), num1 = c(1L, 2L), num2 = c(2, 3.4))

  expect_identical(data_type(test_df)$num, test_df %>% select(num1, num2))
  expect_identical(data_type(test_df)$cat, test_df %>% select(cat1, cat2))
})

test_that("data_type correctly captures input error", {
  expect_error(data_type("test_df")$num, "Please provide a valid data.frame object")
  expect_error(data_type(tibble())$num, "Please provide a non-empty data.frame object")
})

}

test-data_type()
```

**Review by camadi — APPROVED** [review]

**Pull request merged**